src/critics/reference_trajectory_critic.cpp

```diff
@@ -92,13 +99,65 @@ xt::xtensor<double,
           P[0] = P1(s, 0) + u * P2_P1_diff(s, 0);
           P[1] = P1(s, 1) + u * P2_P1_diff(s, 1);
         }
-        min_dist = std::min(evaluate_dist(std::move(P), t, p), min_dist);
+        auto dist = evaluate_dist(std::move(P), t, p);
+        if (dist < min_dist) {
+          min_s = s;
+          min_dist = dist;
+        }
       }
+      max_s = std::max(max_s, min_s);
```

Comment:
Is this signed? I'm not sure `max_s` would ever be not `min_s` otherwise, since `max_s` is never assigned a value after being initialized to `0`

Reply:
I'm not sure i get it right. 
min_s is index of nearest segment to trajectory point, 
max_s is farthest segment point among those nearest segments

Reply:
You're right, I misread